src/Hedgehog/Property.fs

```diff
@@ -278,25 +72,25 @@ module Property =
         Gen.constant (Journal.delayedSingleton msg, Success ()) |> ofGen
 
     let private mapGen
-            (f : Gen<Journal * Result<'a>> -> Gen<Journal * Result<'b>>)
+            (f : Gen<Journal * Outcome<'a>> -> Gen<Journal * Outcome<'b>>)
             (x : Property<'a>) : Property<'b> =
         toGen x |> f |> ofGen
 
     [<CompiledName("Map")>]
     let map (f : 'a -> 'b) (x : Property<'a>) : Property<'b> =
-        (mapGen << Gen.map << second << Result.map) f x
+        (mapGen << Gen.map << Tuple.mapSecond << Outcome.map) f x
 
     let private bindGen
-            (m : Gen<Journal * Result<'a>>)
-            (k : 'a -> Gen<Journal * Result<'b>>) : Gen<Journal * Result<'b>> =
+            (m : Gen<Journal * Outcome<'a>>)
+            (k : 'a -> Gen<Journal * Outcome<'b>>) : Gen<Journal * Outcome<'b>> =
         Gen.bind m <| fun (journal, result) ->
             match result with
             | Failure ->
                 Gen.constant (journal, Failure)
             | Discard ->
                 Gen.constant (journal, Discard)
             | Success x ->
-                Gen.map (first (Journal.append journal)) (k x)
+                Gen.map (Tuple.mapFirst (Journal.append journal)) (k x)
```

Comment:
When I have nested functors like this, I like to define a module for them.
```fsharp
module GenTuple =
  let mapFirst f = f |> Tuple.mapFirst |> Gen.map
```

This helps to reduce the number of matches pars of parentheses, which [I prefer to minimize](https://blog.ploeh.dk/2020/10/19/monomorphic-functors/#bb0bb9d1bf094c1897f9dd42fb49712c).

Reply:
Done